src/librustc_mir/hair/mod.rs

```diff
@@ -28,6 +26,8 @@ use self::cx::Cx;
 
 pub mod cx;
 
+pub use rustc_const_eval::pattern::{BindingMode, Pattern, PatternKind, FieldPattern};
```

Comment:
Another idea would be to move it to `rustc_typeck` (and maybe have HAIR definitions in `rustc`).